test/cpp/codegen/compiler_test_golden

```diff
@@ -105,6 +106,22 @@ class ServiceA final {
       return std::unique_ptr< ::grpc::ClientAsyncReaderWriterInterface< ::grpc::testing::Request, ::grpc::testing::Response>>(PrepareAsyncMethodA4Raw(context, cq));
     }
     // Method A4 trailing comment 1
+    class experimental_async_interface {
```

Comment:
Should we call this experimental instead of experimental_async just to stay consistent with the generic version?

Reply:
Great question. I'd actually suggest going the other way, and naming that experimental_async as well for now. Longer term, the design plan is to just use "async()" as the marker. I had earlier thought that GenericStub could just go without any marker like that, since the overloads are sufficient for distinction, but I think that actually a marker for consistency would be reasonable.

Reply:
In retrospect, I'd prefer to push this to a different PR that focuses again on generic stub, if that's ok with you.

Reply:
Sure